lib/db/namespaces.js

```diff
@@ -0,0 +1,79 @@
+const { fetch } = require('../fetch');
+const FormData = require('form-data');
+const { get: getOptions } = require('./options');
+const lodashGet = require('lodash/get');
+
+const { httpBody } = require('../response-transforms');
+
+// For Stardog <= 7.0.2 compatibility. Remove later.
+const backwardsCompatGetNamespaces = (conn, database, params) =>
+  getOptions(conn, database, params).then(res => {
+    if (res.status === 200) {
+      const n = lodashGet(res, 'body.database.namespaces', []);
+      const names = n.reduce((memo, keyValString) => {
+        const [key, value] = keyValString.split('=');
+        return Object.assign({}, memo, { [key]: value });
+      }, {});
+      res.body = names;
+    }
+    return res;
+  });
+
+const get = (conn, database, params) => {
+  const headers = conn.headers();
+  headers.set('Accept', 'application/json');
+  return fetch(conn.request(database, 'namespaces'), {
+    method: 'GET',
+    headers,
+  }).then(res => {
+    // Old version of Stardog:
+    if (res.status === 404) {
+      return backwardsCompatGetNamespaces(conn, database, params);
+    }
+
+    return httpBody(res).then(bodyRes =>
+      Object.assign({}, bodyRes, {
+        body: bodyRes.body.namespaces.reduce(
+          (memo, { prefix, name }) =>
+            Object.assign({}, memo, { [prefix]: name }),
+          {}
+        ),
+      })
+    );
+  });
+};
+
+const set = (conn, database, fileOrContents, options = {}) => {
```

Comment:
i lean towards `update` if it's not replacing, but also don't mind `set`

Reply:
as long as the docs are clear about this, it doesn't matter much to me

Reply:
I opted to change it to `add`.